app/models/miq_server/worker_management/monitor/kill.rb

```diff
@@ -4,14 +4,37 @@ module MiqServer::WorkerManagement::Monitor::Kill
   def kill_all_workers
     return unless self.is_local?
 
-    killed_workers = []
-    miq_workers.each do |w|
-      if MiqWorker::STATUSES_CURRENT_OR_STARTING.include?(w.status)
-        w.kill
-        worker_delete(w.pid)
-        killed_workers << w
+    remove_unknown_workers
+    remove_workers
+  end
+
+  private
+
+  def remove_unknown_workers
+    # We try, but forget to write migrations to delete orphaned worker class rows, causing this method
+    # to blow up and cause the server to continually fail to start.
+    # This is called very early from evm_application and ensures we attempt to call MiqWorker#kill
+    # to remove the problematic rows, and their in flight queue messages.
+    bad_workers = miq_workers.where.not(:type => MiqWorker.descendants.map(&:name))
```

Comment:
I hadn't looked at this PR since you originally created it. I like this approach much better than what you originally had 👏